docs/react-testing-library/example-intro.md

```diff
@@ -4,25 +4,77 @@ title: Example
 sidebar_label: Example
 ---
 
-## Full Example
+## Basic Example
 
-See the following sections for a detailed breakdown of the test
+Here is our first example, a simple component that changes a counter text when a
+user clicks a button:
+
+```jsx
+// Counter.js
+
+import React, { useState } from 'react'
+
+function Counter() {
+  const [count, setCount] = useState(0)
+
+  return (
+    <div>
+      <p>Times clicked: {count}</p>
+      <button onClick={() => setCount(count + 1)}>Click me</button>
+    </div>
+  )
+}
+
+export default Counter
+```
+
+```js
+// Counter.test.js
+
+// import dependencies
+import React from 'react'
+import ReactDOM from 'react-dom'
+
+// import react-testing methods
+import { render, fireEvent } from '@testing-library/react'
+
+// import the component to test
+import Counter from '../Counter.js'
+
+test('increments value on click', () => {
+  // The render method returns a collection of utilities to query your component
+  const { getByText } = render(<Counter />)
+
+  // getByText returns the first matching node for a query, and throws an error
+  // if no elements match or if more than one match is found
+  getByText('Times clicked: 0')
+
+  const button = getByText('Click me')
+
+  // Fires a click DOM event to the specified target
+  fireEvent.click(button)
+  fireEvent.click(button)
+
+  getByText('Times clicked: 2')
```

Comment:
I prefer to use case-insensitive regex rather than strings because they're more resilient to changes. Could this (and others like it) be changed to: `getByText(/times clicked: 2/i)`